src/vm.rs

```diff
@@ -1007,14 +1007,19 @@ impl<'a, T: GuestStagePagingMode> FinalizedVm<'a, T> {
     ) -> EcallAction {
         use NaclFunction::*;
         match nacl_func {
-            SetShmem { shmem_pfn } => self.set_shmem_area(shmem_pfn, active_vcpu).into(),
+            ProbeFeature { feature_id } => self.probe_nacl_feature(feature_id).into(),
+            SetShmem { shmem_addr } => self.set_shmem_area(shmem_addr, active_vcpu).into(),
         }
     }
 
-    fn set_shmem_area(&self, shmem_pfn: u64, active_vcpu: &mut ActiveVmCpu<T>) -> EcallResult<u64> {
-        if shmem_pfn != u64::MAX {
+    fn set_shmem_area(
+        &self,
+        shmem_addr: u64,
+        active_vcpu: &mut ActiveVmCpu<T>,
+    ) -> EcallResult<u64> {
+        if shmem_addr != u64::MAX {
             // Pin the pages that the VM wants to use for the shared state buffer.
-            let shared_page_addr = self.guest_addr_from_raw(shmem_pfn << PFN_SHIFT)?;
+            let shared_page_addr = self.guest_addr_from_raw(shmem_addr)?;
```

Comment:
Add a sanity check for the 4K alignment now ?

Reply:
`guest_addr_from_raw` is doing the alignment check.